lib/sift/client.rb

```diff
@@ -165,91 +220,215 @@ def track(event, properties = {}, timeout = nil, path = nil, return_score = fals
 
       raise("user_id must be a non-empty string") if (!user_id.is_a? String) || user_id.to_s.empty?
-      timetout ||= @timeout
 
-      options = { :headers => {"User-Agent" => user_agent} }
+      query = {}
+      query[:api_key] = api_key
+      query[:abuse_type] = abuse_type if abuse_type
+
+      options = {
+        :headers => {},
+        :query => query
+      }
+      options.merge!(:timeout => timeout) unless timeout.nil?
+
+      response = self.class.delete(Sift.users_label_api_path(user_id, version), options)
+      Response.new(response.body, response.code, response.response)
+    end
+
+
+    # Gets the status of a workflow run.
+    #
+    # See https://siftscience.com/developers/docs/ruby/workflows-api/workflow-status .
+    #
+    # ==== Parameters
+    #
+    # run_id::
+    #   The ID of a workflow run.
+    #
+    # opts (optional)::
+    #   A Hash of optional parameters for this request.
+    #
+    def get_workflow_status(run_id, opts = {})
+      account_id = opts[:account_id] || @account_id
+      api_key = opts[:api_key] || @api_key
+      timeout = opts[:timeout] || @timeout
+
+      options = {
+        :headers => { "User-Agent" => user_agent },
+        :basic_auth => { :username => api_key, :password => "" }
+      }
       options.merge!(:timeout => timeout) unless timeout.nil?
-      path = Sift.current_users_label_api_path(user_id)
-      response = self.class.delete(path + "?api_key=#{@api_key}", options)
+
+      uri = API3_ENDPOINT + Sift.workflow_status_path(account_id, run_id)
+      response = self.class.get(uri, options)
       Response.new(response.body, response.code, response.response)
     end
 
+
+    # Gets the decision status of a user.
+    #
+    # See https://siftscience.com/developers/docs/ruby/decisions-api/decision-status .
+    #
+    # ==== Parameters
+    #
+    # user_id::
+    #   The ID of user.
+    #
+    # opts (optional)::
+    #   A Hash of optional parameters for this request.
+    #
+    def get_user_decisions(user_id, opts = {})
+      account_id = opts[:account_id] || @account_id
+      api_key = opts[:api_key] || @api_key
+      timeout = opts[:timeout] || @timeout
+
+      options = {
+        :headers => { "User-Agent" => user_agent },
+        :basic_auth => { :username => api_key, :password => "" }
+      }
+      options.merge!(:timeout => timeout) unless timeout.nil?
+
+      uri = API3_ENDPOINT + Sift.user_decisions_api_path(account_id, user_id)
+      response = self.class.get(uri, options)
+      Response.new(response.body, response.code, response.response)
+    end
+
+
+    # Gets the decision status of an order.
+    #
+    # See https://siftscience.com/developers/docs/ruby/decisions-api/decision-status .
+    #
+    # ==== Parameters
+    #
+    # order_id::
+    #   The ID of an order.
+    #
+    # opts (optional)::
+    #   A Hash of optional parameters for this request.
+    #
+    def get_order_decisions(order_id, opts = {})
```

Comment:
There's value in the concrete clarity of the specific methods, which offsets the redundancy.  Also, `get_decisions_for_entity` would be very easy to add in backward-compatible way in the future.  It may be worthwhile to make that a base method that both call into now, or it could be done later... not too big a deal.